internal/codegen/dir.go

```diff
@@ -0,0 +1,66 @@
+package codegen
+
+import (
+	"io"
+	"path/filepath"
+	"time"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/sirupsen/logrus"
+	"go.rgst.io/stencil/internal/modules"
+	"sigs.k8s.io/yaml"
+)
+
+// rawDirManifest is a helper to unmarshal data from a directory manifest yaml
+type rawDirManifest struct {
+	RenameTemplate string `yaml:"renameTemplate"`
+}
+
+// DirManifest encompasses data from optional directory-based manifests
+type DirManifest struct {
+	OriginalPathBase string
+
+	Template *Template
+
+	ReplaceName string
+}
+
+// LoadDirManifest unmarshals a directory manifest from a directory
+func LoadDirManifest(fs billy.Filesystem, path string, mod *modules.Module, log logrus.FieldLogger) (*DirManifest, error) {
+	f, err := fs.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	conts, err := io.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+
+	var rm rawDirManifest
+	if err := yaml.Unmarshal(conts, &rm); err != nil {
+		return nil, err
+	}
+
+	rt, err := NewTemplate(mod, path, 0o000, time.Time{}, []byte(rm.RenameTemplate), log)
```

Comment:
I'm worried that allowing the dir manifest to be templated will bring it's own set of issues.  I'm wondering if we could do something like this instead:

```yaml
outputDir: '{{ .Config.Name }}'
```
The ideas is this would replace the entire output directory.

Reply:
Slight problems with that because it'd require that we know what part of it to replace. That got me thinking about support variables in the directory string. Which made me realize, why don't we just support that in the actual directory?

e.g., `cmd/{{ .Config.Name}}/app.go.tpl`?

Reply:
For context, I want to avoid _too_ much flexibility unless it really makes sense to do so because it's much harder to remove things later than add them.

Reply:
A caveat would be that globals can't control it, but I'm concerned about allowing other modules to somehow have side effects to that extent... 😓

Reply:
because that dir name won't work on windows, and prolly other terrible filesystems

Reply:
(i thought about that tho)

it also needs to be able to read globals in our setup, not just manifest params, which come in after the first render pass, but same idea

Reply:
I'm still pretty weary about templating an entire config file first-class upfront. I'm not a huge fan of it, but what do you think about this?

```yaml
# Transforms 'cmd/$packageName' -> 'cmd/packageName'
outputDirVars:
  packageName: '{{ stencil.GetGlobal "aThing" }}'
```

It's assuming we can access globals here, but I think we could do that pretty safely.

Reply:
We settled on not doing that because of possible file system issues `$` could present 😢